### DISCUSSION TITLE
Problem with pull

### DISCUSSION THREAD

**Discussion by faureunige** [munoztd0/DAP_2021#17] [Show and tell]
Hi ! When i try to pull, I have this error messge : what can I do ? 

>>> /usr/local/bin/git pull
error: Your local changes to the following files would be overwritten by merge:
	.DS_Store
Please commit your changes or stash them before you merge.
Aborting
Updating de190cf..271fa29

**Comment by munoztd0** [comment]
Have you committed all your changes ? If yes che k and see if they are boxes with blue squares on the commit panel, if yes click on them so they change to circles and commit and pull again.

**Reply by faureunige** [reply]
seems to be solved, thanks !

**Reply by munoztd0** [reply]
🙌 you're welcome, keep up the good work and have a nice new years eve

**Reply by faureunige** [reply]
well, another problem actually, sorry haha ! When i try to knit to PDF, I get : 


output file: Marketing_analysis_Markdown.knit.md

Erreur : 'set_envvar' is not an exported object from 'namespace:xfun'
Exécution arrêtée

**Reply by munoztd0** [reply]
![Screenshot_20211230-225109](https://user-images.githubusercontent.com/43644805/147790297-613e1a67-49ee-4a38-a865-a94b3fcf928b.png)
Try updating xfun, knitr and rmarkdown and see how it goes . (To do that go into packages -> click on the circle arrow (update) and then you can choose what to update). PS: you might want to update all your package at the same BUT know that might take a while so do it when you are finished working so you can let it do it's things for an hour or two)

**Comment by faureunige** [comment]
![Capture d’écran 2021-12-31 à 09 00 49](https://user-images.githubusercontent.com/82580274/147811100-bb2f2d9a-7be4-4ee8-86b1-ccd9a530cb0a.png)
![Capture d’écran 2021-12-31 à 09 01 08](https://user-images.githubusercontent.com/82580274/147811108-127b2231-0e13-4442-a6cb-44a573a0cefa.png)

**Reply by faureunige** [reply]
I updated all the packages, thanks ! I have this message now

**Reply by munoztd0** [reply]
Google is your friend https://www.programmingr.com/r-error-messages/error-in-contrib-urlrepos-source-r-markdown/
And maybe don't load rmarkdown in pload().

**Reply by faureunige** [reply]
I did what is said on this website, but i still return to the same problem (really sorry for the spamming ! I saw we have to submit a Rmd draft until today so I'm a bit stressed out...) :

! sh: pdflatex: command not found

Erreur : LaTeX failed to compile Marketing_analysis_Markdown.tex. See https://yihui.org/tinytex/r/#debugging for debugging tips. See Marketing_analysis_Markdown.log for more info.
De plus : Message d'avis :
Dans system2(..., stdout = if (use_file_stdout()) f1 else FALSE, stderr = f2) :
  erreur lors de l'exécution d'une commande
Exécution arrêtée

No LaTeX installation detected (LaTeX is required to create PDF output). You should install a LaTeX distribution for your platform: https://www.latex-project.org/get/

  If you are not sure, you may install TinyTeX in R: tinytex::install_tinytex()

  Otherwise consider MiKTeX on Windows - http://miktex.org

  MacTeX on macOS - https://tug.org/mactex/
  (NOTE: Download with Safari rather than Chrome _strongly_ recommended)

  Linux: Use system package manager

**Reply by faureunige** [reply]
(So I'm downloading MacTex now)

**Reply by munoztd0** [reply]
Yeah that should do it @danielsgranja  had a similar/ same issue ?

You can also knit your rmarkdown to html instead of pdf that could also work..

But don't worry too much about this deadline, it is just for a "draft"

**Reply by munoztd0** [reply]
How are you doing?

**Reply by faureunige** [reply]
Hi, fine thanks ! I just have a few questions ; I see that my html file is 25 pages, which seems way too long - is there a way to select which commands are executed ? (when i knit, it executes everything, and the figures take a lot of space)

**Reply by faureunige** [reply]
Also, for PCA, most of my interest variables are factors - I am still hesitating if it really makes sense to do a PCA on this data then...

**Reply by munoztd0** [reply]
You can do whatever you want,  checkout The R Markdown Cheat Sheet http://blog.rstudio.org/2014/08/01/the-r-markdown-cheat-sheet/


nor run them by 
`
```{r, eval = FALSE} 
`



hide stuff  by
![image](https://user-images.githubusercontent.com/43644805/147937116-07e41084-c298-4258-ae18-e761bd62cd48.png)

or make them smaller:
`
```{r annexes, fig.show="hold", out.width="50%", fig.align = 'center'}
`

Here are examples: https://github.com/munoztd0/DAP_2021/blob/main/RESOURCES/Rmardkown_tips_and_tricks.Rmd 
let me know if something doesn't make sense..

Are you able to push your stuff now ? I don't see any html file yet

**Reply by munoztd0** [reply]
For the PCA don't worry too much about what "make the most sense" just try to do it and see where it goes.. The results is not the most relevant, but the journey is here !

And you have plenty of numerical variables such as: Age,  MntMeatProducts, MntFishProducts,	MntSweetProducts,	MntGoldProds,	NumDealsPurchases, NumWebPurchases, NumCatalogPurchases, NumStorePurchases, NumWebVisitsMonth.

You can try plenty of stuff with those I think, but lookout for very correlated variables. And don't worry if you don't find "interesting" or "signifivative" stuff, be transparent, be serious, be pragmatic, tell what you found, what you didn't find and that's it, that's stats, that's science..

**Reply by faureunige** [reply]
amazing thank you so much ! one other question, for NumStorePurchases, the skewness (0,7) doesn't change when i use sqrt or not. And in both cases, the distribution is not very normal... can I justify using lm by saying that skewness is low, even thought the variable is not really normal ?

**Reply by munoztd0** [reply]
Show me pictures (just screenshots them into here)

**Reply by faureunige** [reply]
![Capture d’écran 2022-01-03 à 18 51 17](https://user-images.githubusercontent.com/82580274/147962908-2b5f0079-6cb9-42f2-a493-e1f1ace021cb.png)
![Capture d’écran 2022-01-03 à 18 51 27](https://user-images.githubusercontent.com/82580274/147962913-99a20ec8-4e10-4071-9845-31999bfa8f6a.png)

**Reply by munoztd0** [reply]
Go with it (the regular not the sqrt)

**Reply by faureunige** [reply]
Hi, I have one last question about PCA, when I type 
clean_data <-data[rowSums(is.na(data))==0, ]
pm1<-prcomp(clean_data[,-c(1,3,4,6,7,8,18)],  scale=TRUE)
data2 <- cbind(data, pm1$x)
str(data2)

I get 
Erreur dans data.frame(..., check.names = FALSE) : 
  les arguments impliquent des nombres de lignes différents : 2240, 2212 


After this is solved, I will submit the draft, since I'm done.

**Reply by munoztd0** [reply]
When you have an error, go line by line to identify the error location. 

Here it seems that :
data2 <- cbind(data, pm1$x) should instead be : 
data2 <- cbind(clean_data, pm1$x) right ?

PS: I modified a few things on you rrmakdown so try to pull

**Reply by faureunige** [reply]
thank you for the modifications. Apart from the missing comments, that i am adding now, am I still missing anything fundamental ? or is the analysis complete ?

**Reply by munoztd0** [reply]
No nothing fundamental, but a good conclusion and as you said more explanation, comments
